CSharpFunctionalExtensions/Result/Extensions/TapAsyncBoth.cs

```diff
@@ -69,5 +69,35 @@ public static async Task<Result<T>> Tap<T>(this Task<Result<T>> resultTask, Func
 
             return result;
         }
+
+        /// <summary>
+        ///     Executes the given action if the calling result is a success. Returns the calling result on success.
+        /// </summary>
+        public static async Task<Result> Tap<T>(this Task<Result<T>> resultTask, Func<T, Task<Result>> func)
```

Comment:
We need to remove this one

Reply:
> **EDITED**
> I think we are missing this signature:
> ```cs
> public static Result<T> Tap<T>(this Result<T> result, Func<T, Result> func)
> {
>     if (result.IsFailure)
>         return result;
> 
>     return func(result.Value).Map(() => result.Value);
> }
> ```
> 
Shouldn't all the overloads have a `Func` that returns simply a `Result`:

Are we also missing these then?
```
Task<Result<T>> Tap<T>(this Task<Result<T>> resultTask, Func<T, Task<Result>> func)
Task<Result<T>> Tap<T>(this Task<Result<T>> resultTask, Func<T, Result> func)
Task<Result<T>> Tap<T>(this Result<T> result, Func<T, Task<Result>> func)
```

Reply:
Because it doesn't return `Task<Result<T>>` ? Okay, removing.